torchgeometry/image/laplacian.py

```diff
@@ -0,0 +1,169 @@
+        ksize (int): filter size should be odd.
+
+    Returns:
+        Tensor: 2D tensor with laplacian filter matrix coefficients.
+
+    Shape:
+        - Output: :math:`(ksize, ksize)`
+
+    Examples::
+
+        >>> tgm.image.get_laplacian_kernel2d(3)
+        tensor([[ 1.,  1.,  1.],
+        [ 1., -8.,  1.],
+        [ 1.,  1.,  1.]])
+
+        >>> tgm.image.get_laplacian_kernel2d(5)
+        tensor([[  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1., -24.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.]])
+
+    """
+    if not isinstance(ksize, int) or ksize % 2 == 0 or ksize <= 0:
+        raise TypeError("ksize must be a tuple of length two. Got {}"
+                        .format(ksize))
+
+    kernel = torch.ones((ksize, ksize))
+    mid = int(math.floor((ksize / 2)))
+    kernel[mid, mid] = 1 - math.pow(ksize, 2)
+    kernel_2d: torch.Tensor = kernel
+    return kernel_2d
+
+
+class LaplacianBlur(nn.Module):
+    r"""Creates an operator that blurs a tensor using a Laplacian filter.
+
+    The operator smooths the given tensor with a laplacian kernel by convolving
+    it to each channel. It suports batched operation.
+
+    Arguments:
+        kernel_size (int): the size of the kernel.
+
+    Returns:
+        Tensor: the blurred tensor.
+
+    Shape:
+        - Input: :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)`
+
+    Examples::
+
+        >>> input = torch.rand(2, 4, 5, 5)
+        >>> laplace = tgm.image.LaplacianBlur(5)
+        >>> output = laplace(input)  # 2x4x5x5
+    """
+
+    def __init__(self, kernel_size: int) -> None:
+        super(LaplacianBlur, self).__init__()
+        self.kernel_size: int = kernel_size
+        self._padding: int = self.compute_zero_padding(kernel_size)
+        self.kernel: torch.Tensor = self.get_laplacian_kernel(kernel_size)
+
+    @staticmethod
+    def get_laplacian_kernel(kernel_size) -> torch.Tensor:
+        """Returns a 2D Laplacian kernel array."""
+        kernel: torch.Tensor = get_laplacian_kernel2d(kernel_size)
+        return kernel
+
+    @staticmethod
+    def compute_zero_padding(kernel_size: int):
+        """Computes zero padding."""
+        computed = (kernel_size - 1) // 2
+        return computed
+
+    def forward(self, x: torch.Tensor):
+        if not torch.is_tensor(x):
+            raise TypeError("Input x type is not a torch.Tensor. Got {}"
+                            .format(type(x)))
+        if not len(x.shape) == 4:
+            raise ValueError("Invalid input shape, we expect BxCxHxW. Got: {}"
+                             .format(x.shape))
+        # prepare kernel
+        b, c, h, w = x.shape
+        tmp_kernel: torch.Tensor = self.kernel.to(x.device).to(x.dtype)
+        kernel: torch.Tensor = tmp_kernel.repeat(c, 1, 1, 1)
+
+        # convolve tensor with gaussian kernel
+        return conv2d(x, kernel, padding=self._padding, stride=1, groups=c)
```

Comment:
mark as a todo to implement filter2D: https://github.com/opencv/opencv/blob/7fb70e170154d064ef12d8fec61c0ae70812ce3d/modules/imgproc/src/deriv.cpp#L822

Reply:
Yes, indeed! We should also look at OpenCV's separable filter when actually implementing a generic filter function.